src/Silverback.Integration/Messaging/Validation/MessageValidator.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2020 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+
+namespace Silverback.Messaging.Validation
+{
+    /// <summary>
+    ///     Validates the message and throws a <see cref="MessageValidationException" /> depending on the
+    ///     <see cref="MessageValidationMode" />.
+    /// </summary>
+    internal static class MessageValidator
+    {
+        /// <summary>
+        ///     Validates the message.
+        /// </summary>
+        /// <param name="message">The message.</param>
+        /// <param name="validationMode">The message validation mode.</param>
+        /// <returns>
+        ///     A Tuple indicating whether the message is valid and the validation errors.
+        /// </returns>
+        /// <exception cref="MessageValidationException">
+        ///     The thrown exception when validation mode is
+        ///     <see cref="MessageValidationMode.ThrowException" />.
+        /// </exception>
+        public static (bool IsValid, string? ValidationErrors) CheckMessageIsValid(
+            object message,
+            MessageValidationMode validationMode)
+        {
+            var validationContext = new ValidationContext(message);
+            var results = new List<ValidationResult>();
+            var validMessage = Validator.TryValidateObject(message, validationContext, results, true);
+
+            if (validMessage)
+            {
+                return (true, null);
+            }
+
+            var validationResults = string.Join(
+                Environment.NewLine,
+                results.Select(validationResult => validationResult.ErrorMessage));
+
+            if (validationMode == MessageValidationMode.ThrowException)
+            {
+                throw new
+                    MessageValidationException($"Message not valid. {validationResults}");
```

Comment:
Is the message formatted correctly?

I would like that it looked like:
```
The message is not valid:
- FirstName is too long
- Email doesn't match the regex
- Age cannot be a negative number
```

Would that be possible?

Reply:
Right! I fixed it!